src/main/kotlin/no/nav/medlemskap/regler/funksjoner/ArbeidsforholdFunksjoner.kt

```diff
@@ -143,6 +143,42 @@ object ArbeidsforholdFunksjoner {
         return Math.round(vektetStillingsprosentForArbeidsforhold * 10.0) / 10.0
     }
 
+    fun List<Arbeidsforhold>.harBrukerJobbetMerEnnGittStillingsprosentTilEnhverTidSkygge(gittStillingsprosent: Double, kontrollPeriode: Kontrollperiode): Boolean {
+        val arbeidsforholdForKontrollPeriode = this.arbeidsforholdForKontrollPeriode(kontrollPeriode)
+        var samletStillingsprosent = 0.0
+
+        for (arbeidsforhold in arbeidsforholdForKontrollPeriode) {
+            val vektetStillingsprosentForArbeidsforhold = arbeidsforhold.vektetStillingsprosentForArbeidsforholdSkygge(kontrollPeriode)
+
+            if (vektetStillingsprosentForArbeidsforhold < gittStillingsprosent &&
+                this.arbeidsforholdForKontrollPeriode(kontrollPeriode).ingenAndreParallelleArbeidsforhold(arbeidsforhold)
+            ) {
+                return false
+            }
+
+            samletStillingsprosent += vektetStillingsprosentForArbeidsforhold
+        }
+
+        return samletStillingsprosent >= gittStillingsprosent
+    }
+
+    private fun Arbeidsforhold.vektetStillingsprosentForArbeidsforholdSkygge(kontrollPeriode: Kontrollperiode, beregnGjennomsnittForKontrollperioden: Boolean = false): Double {
+
+        val arbeidsforholdKontrollperiodeIntersection = this.periode.intersection(kontrollPeriode)
+        val totaltAntallDager = if (beregnGjennomsnittForKontrollperioden) kontrollPeriode.antallDager
+        else arbeidsforholdKontrollperiodeIntersection.antallDager
+
+        var vektetStillingsprosentForArbeidsforhold = 0.0
+
```

Comment:
Bra hvis du greier å skrive denne vektingen mer funksjonelt, istedet for at du må ha en var og en for-løkke.

Reply:
Skal gjøre et forsøk! Merger sålenge for å se resultatet av det i Grafana